cmd/channel/webhook/main.go

```diff
@@ -0,0 +1,149 @@
+			},
+			Default:  "POST",
+			Required: true,
+			Options: map[string]string{
+				// https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods
+				"GET":     "GET",
+				"HEAD":    "HEAD",
+				"POST":    "POST",
+				"PUT":     "PUT",
+				"DELETE":  "DELETE",
+				"CONNECT": "CONNECT",
+				"OPTIONS": "OPTIONS",
+				"TRACE":   "TRACE",
+				"PATCH":   "PATCH",
+			},
+		},
+		{
+			Name: "url_template",
+			Type: "string",
+			Label: map[string]string{
+				"en_US": "URL Template",
+				"de_DE": "URL-Template",
+			},
+			Help: map[string]string{
+				"en_US": "URL, optionally as a Go template over the current plugin.NotificationRequest.",
+				"de_DE": "URL, optional als Go-Template über das zu verarbeitende plugin.NotificationRequest.",
+			},
+			Required: true,
+		},
+		{
+			Name: "request_body_template",
+			Type: "string",
+			Label: map[string]string{
+				"en_US": "Request Body Template",
+				"de_DE": "Anfragedaten-Template",
+			},
+			Help: map[string]string{
+				"en_US": "Go template applied to the current plugin.NotificationRequest to create an request body.",
+				"de_DE": "Go-Template über das zu verarbeitende plugin.NotificationRequest zum Erzeugen der mitgesendeten Anfragedaten.",
+			},
+			// Default as a minimal working example to send a JSON object.
+			Default: `{{json .}}`,
+		},
+	}
+
+	configAttrs, err := json.Marshal(elements)
+	if err != nil {
+		panic(err)
+	}
+
+	return &plugin.Info{
+		Name:             "Webhook",
+		Version:          internal.Version.Version,
+		Author:           "Icinga GmbH",
+		ConfigAttributes: configAttrs,
+	}
+}
+
+func (ch *Webhook) SetConfig(jsonStr json.RawMessage) error {
+	err := json.Unmarshal(jsonStr, ch)
+	if err != nil {
+		return err
+	}
+
+	tmplFuncs := template.FuncMap{
+		"json": func(a any) (string, error) {
+			data, err := json.Marshal(a)
+			if err != nil {
+				return "", err
+			}
+			return string(data), nil
+
+		},
+	}
+
+	ch.tmplUrl, err = template.New("url").Funcs(tmplFuncs).Parse(ch.URLTemplate)
+	if err != nil {
+		return fmt.Errorf("cannot parse URL template: %w", err)
+	}
+
+	ch.tmplRequestBody, err = template.New("request_body").Funcs(tmplFuncs).Parse(ch.RequestBodyTemplate)
+	if err != nil {
+		return fmt.Errorf("cannot parse Request Body template: %w", err)
+	}
+
+	return nil
+}
+
+func (ch *Webhook) SendNotification(req *plugin.NotificationRequest) error {
+	var urlBuff, reqBodyBuff bytes.Buffer
+	if err := ch.tmplUrl.Execute(&urlBuff, req); err != nil {
+		return fmt.Errorf("cannot execute URL template: %w", err)
+	}
+	if err := ch.tmplRequestBody.Execute(&reqBodyBuff, req); err != nil {
+		return fmt.Errorf("cannot execute Request Body template: %w", err)
+	}
+
+	httpReq, err := http.NewRequest(ch.Method, urlBuff.String(), &reqBodyBuff)
+	if err != nil {
+		return err
+	}
+	httpResp, err := http.DefaultClient.Do(httpReq)
+	if err != nil {
+		return err
+	}
+
+	_, _ = io.Copy(io.Discard, httpResp.Body)
+	_ = httpResp.Body.Close()
+
+	return nil
```

Comment:
This should probably perform some check on the response, at least that the status code was 2xx or something like that.

Reply:
When your endpoint honors HTTP, then yes. But in our sad reality, this mustn't be the case. Unless one wants to exclude someone's potential use case, then this must be configurable.

However, this opens a wide field of optional configurations: additional or changed HTTP headers, HTTP versions, TLS configuration, response parsing and whatnot. At day's end, one might end up with something like the good old [check_http plugin](https://www.monitoring-plugins.org/doc/man/check_http.html) - can do variants of everything, is hard to test and maybe one just wanna use a wrapper around curl anyway.

I don't want to create a straw man against performing a small change, but want to underline both that I considered the idea and that making a channel (or check plugin) as flexible as a protocol like HTTP is maybe too much. And then there are even further cases, e.g., `PUT`ing a new Notification into a very REST-ful backend and `DELETE`ing when it's resolved (_inb4: just create a custom channel_).

Reply:
> When your endpoint honors HTTP, then yes. But in our sad reality, this mustn't be the case. Unless one wants to exclude someone's potential use case, then this must be configurable.

What else should they speak but HTTP? I mean this isn't about allowing for every possible scenario like always sending 200 OK but then including an error message in the response body. With the current state of the PR, there is no reasonable way for the remote to signal an error, like they would have to close the connection without sending a response to do so.

Reply:
Speaking enough HTTP to make a client happy and honoring the HTTP specifications are two different things, I'd say. But yes, I am going to add an expected status code, defaulting to 200.

I mainly wanted to express my fear of this simple channel degenerating into some untestable state monster that honors every use case while being horrible to use.